src/generic/graphicc.cpp

```diff
@@ -2360,6 +2378,14 @@ wxCairoContext::~wxCairoContext()
 
 void wxCairoContext::Init(cairo_t *context)
 {
+#ifdef __WXGTK__
+    // Attempt to find the system font scaling parameter (e.g. "Fonts->Scaling
+    // Factor" in Gnome Tweaks, "Force font DPI" in KDE System Settings or
+    // GDK_DPI_SCALE environment variable).
+    GdkScreen* screen = gdk_screen_get_default();
```

Comment:
Hm, is this correct for gtk2? Shouldn't this be done only for gtk3 or newer?
I'm using gtk2 on a monitor with dpi of 16x and I've not seen small fonts on wx+gtk2, but the same application built with wx+gtk3 is totally broken (wxtreectrl, wxpropgrid, etc have small fonts).

I can do a test with both gtk2 and gtk3. Which sample should I test with?

Reply:
This is a good point, AFAIK GTK 2 indeed doesn't use fractional font scaling, so we should probably make all these `#ifdef`s check for `__WXGTK3__` instead. @paulcor Could you please confirm this?

I think the best would be to test the widgets sample, as it has both native and generic controls. The idea is that they should use the font of the same size now, whereas it wasn't necessarily the case before (with GTK 3, anyhow).

Reply:
The size seems fine in both gtk2 and gtk3 with this pr. I've tried the widgets and the treectrl samples.
But it seems there is a regression in the treectrl sample. In it the root item is set to have italic font. With this patch this property is lost.

See the attached image for an example. The window with the smaller font is built from commit 9b43bd8af26c91ba779b742e34c1ef089c03e131

![wxtree_sample_gtk3_problem](https://user-images.githubusercontent.com/2459214/68810088-fa63b480-0675-11ea-931d-94704ca73bb7.png)

Also it seems the tree control has some redraw problems when built with GTK3. I see this both with and without the patch. The problem happens when you expand some items and scroll down... I guess I have to open another ticket about this.

Reply:
Thanks for testing, but I can't reproduce the problem with the "Root" item of the treectrl sample. Which version of GTK do you use? Mine is 3.24.12 and the root item does appear in italics, both by default and with `GDK_DPI_SCALE=1.5` (it also becomes larger in this case).

Maybe I inadvertently fixed this while fixing the [other bug](https://github.com/wxWidgets/wxWidgets/pull/1650#discussion_r346975721)? Could you please check if you still see the problem with the latest version of this PR?

Also, it looks like this is never going to do anything for GTK 2, so we could as well not compile this code for it at all.

Reply:
gtk - 2.24.32 and 3.24.11
pango - 1.42.4

And yes, you've fixed it with the latest changes, so this is no longer a problem. :)

Do you see the scrolling problems?

Reply:
> And yes, you've fixed it with the latest changes, so this is no longer a problem. :)

Thanks for testing!
 
> Do you see the scrolling problems?

Yes, it's completely broken :-( I didn't have time to look into it yet, just created [a ticket for it](https://trac.wxwidgets.org/ticket/18584)